runtime/src/lib.rs

```diff
@@ -168,32 +168,43 @@ parameter_types! {
 }
 
 /// Filters and extracts `data` from `call` if it is a `DataAvailability::submit_data` type.
-///
-/// # TODO
-/// - Support utility pallet containing severals `DataAvailability::submit_data` calls and
-/// double-check if we can remove the `clippy::collapsible_match` then.
 impl submitted_data::Filter<RuntimeCall> for Runtime {
-	fn filter(call: RuntimeCall, metrics: submitted_data::RcMetrics) -> Option<Vec<u8>> {
+	fn filter(call: RuntimeCall, metrics: submitted_data::RcMetrics) -> Option<Vec<Vec<u8>>> {
 		metrics.borrow_mut().total_extrinsics += 1;
 
-		#[allow(clippy::collapsible_match)]
 		match call {
-			RuntimeCall::DataAvailability(method) => match method {
-				da_control::Call::submit_data { data } => {
-					let mut metrics = metrics.borrow_mut();
-					metrics.data_submit_leaves += 1;
-					metrics.data_submit_extrinsics += 1;
-					Some(data.into_inner())
-				},
-				_ => None,
+			RuntimeCall::DataAvailability(da_control::Call::submit_data { data }) => {
+				let mut metrics = metrics.borrow_mut();
+				metrics.data_submit_leaves += 1;
+				metrics.data_submit_extrinsics += 1;
+				Some(vec![data.into_inner()])
 			},
-			RuntimeCall::Utility(_) => {
-				// TODO Support utility here.
-				None
+			RuntimeCall::Utility(pallet_utility::Call::batch { calls })
+			| RuntimeCall::Utility(pallet_utility::Call::batch_all { calls })
+			| RuntimeCall::Utility(pallet_utility::Call::force_batch { calls }) => {
+				Self::process_calls(calls, &metrics)
 			},
 			_ => None,
 		}
 	}
+
+	/// This function processes a list of calls and returns their data as Vec<Vec<u8>>
+	fn process_calls(
+		calls: Vec<RuntimeCall>,
+		metrics: &submitted_data::RcMetrics,
+	) -> Option<Vec<Vec<u8>>> {
+		let mut result = Vec::new();
```

Comment:
nit: Reduce possible reallocations
```suggestion
		let mut result = Vec::with_capacity(calls.len());
```